Some Docs suggestions (#114)
thautwarm/MLStyle.jl: docs/tutorials/query-lang.md

Write You A Query Language
===============================================

**P.S**: *this document is not up-to-date*.

You may have heard of embedded query languages like LINQ or extension methods before.

In terms of Julia ecosystem, there is already Query.jl, LightQuery.jl, DataFramesMeta.jl, etc. These packages accomplish the partial or full features of a query language.

> This tutorial primarily shows the creation a concise and efficient query language implementation using MLStyle.jl. This demonstration illustrates the power of MLStyle.jl's ability to perform AST manipulations. Additionally, I think this tutorial can be also extremely helpful to those who're developing query languages for Julia.

Reviewer: ```suggestion
This tutorial primarily shows the creation a concise and efficient query language implemented  with MLStyle.jl. This demonstration illustrates the power of MLStyle.jl's ability to perform AST manipulations. Additionally, I think this tutorial can be also extremely helpful to those who're developing query languages for Julia.
```

Author: Feel like either of these could work, but lets go with yours :)

end
```

> I think it perfect, so let's go ahead. The reason why we make an inline function will be given later. I will disclose that it's used for type inference.

Reviewer: ```suggestion
I think it perfect, so let's go ahead. We'll explain more about why we use `@inline` later, but the short answer is that it was needed for type inference.
```

So, what should the output field names and the source be?

An implementation could be,

```julia
IN_FIELDS, values(GROUPS)
```

> But if so, we will lose the information of group keys, which is not good.

Reviewer: ```suggestion
But if so, we will lose the information of group keys, which is undesirable.
```

```

I can promise you that, each column of your data frames is a `Vector{Any}`, yes, not its actual type.
You may prefer to calculate the type of a column using the common super type of all elements, but there are

> 2 problems if you try this:

Reviewer: ```suggestion
two problems if you try this:
```

- If the column is empty, emmmm...

> - Calculating the super type of all elements incurs an unaffordable cost!

Reviewer: ```suggestion
- Calculating the super type of all elements is very slow!
```

end
```

> You might not be able to understand what the meaning of `fields` and `assigns` are - don't worry - I will explain it for you.

Reviewer: ```suggestion
The meaning of `fields` and `assign`s might not be obvious, so we'll dig closer into these terms now.
```
